src/conversions/infix_to_postfix.rs

```diff
@@ -1,10 +1,16 @@
-// Function to convert infix expression to postfix expression
-pub fn infix_to_postfix(infix: &str) -> String {
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum InfixToPostfixError {
+    UnknownCharacter(char),
+    UnmatchedParent,
+}
+
+/// Function to convert [infix expression](https://en.wikipedia.org/wiki/Infix_notation) to [postfix expression](https://en.wikipedia.org/wiki/Reverse_Polish_notation)
+pub fn infix_to_postfix(infix: &str) -> Result<String, InfixToPostfixError> {
     let mut postfix = String::new();
     let mut stack: Vec<char> = Vec::new();
 
     // Define the precedence of operators
-    let precedence = |op: char| -> i32 {
+    let precedence = |op: char| -> u8 {
```

Comment:
What are the reasons to change from `i32` to `u8`? Just saving these 3 bytes? From the performance point of view `i32` could be faster. I do not have an opinion here, I am just asking.

Reply:
The reason why I changed it, in the first place was the fact that, if someone wants to use this algorithm, he will probably need to adapt it, so while the change doesn't _really_ affect the algorithm, it makes the user understand that there is no negative precedence (AFAIK)